actions.py

```diff
@@ -245,13 +245,20 @@ def validate_time(
 
         def get_duration(entity):
             value = entity.get("value")
-            start = entity.get("from")
-            end = entity.get("to")
-            grain = entity.get("additional_info").get("grain")
-
+            try:
+                start = value.get("from")
+                end = value.get("to")
+                grain = entity.get("additional_info").get("from").get("grain")
+                reportgrain = "timeframe"
+            except:
```

Comment:
can we except a KeyError specifically here? Or whatever exception is being thrown

Reply:
AttributeError -  value will be a string if it's a single time